lib/mix/tasks/tower.test.ex

```diff
@@ -8,5 +8,13 @@ defmodule Mix.Tasks.Tower.Test do
   @impl true
   def run(_args) do
     Tower.test()
+
+    # Avoid automatic BEAM halt after task finishes so we allow time for any
+    # reporters reporting in async processes to finish.
+    Mix.shell().info(
+      "A test exception was generated. You should see it now wherever your configured reporter(s) should report to."
+    )
+
+    Mix.shell().prompt("press ENTER to exit:")
```

Comment:
Alternative tried using just `System.stop` to "wait for a graceful shutdown" of applications, but that wouldn't necessarily mean that async processes reporting would finish before shutdown if not trapping exits.